extendeddatacrossword_test.go

```diff
@@ -24,16 +24,17 @@ type PseudoFraudProof struct {
 }
 
 func TestRepairExtendedDataSquare(t *testing.T) {
+	codec := NewLeoRSCodec()
+
 	tests := []struct {
-		name  string
-		codec Codec
+		name string
 	}{
-		{"leopard", NewLeoRSCodec()},
+		{"leopard"},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
```

Comment:
Great start. I think this test can be simplified even more to remove layers of indentation. For example:

```go
func TestRepairExtendedDataSquare(t *testing.T) {
	codec := NewLeoRSCodec()
	original := createTestEds(codec, shareSize)

	rowRoots, err := original.RowRoots()
	require.NoError(t, err)

	colRoots, err := original.ColRoots()
	require.NoError(t, err)

	// Verify that an EDS can be repaired after the maximum amount of erasures
	t.Run("MaximumErasures", func(t *testing.T) {
		flattened := original.Flattened()
		flattened[0], flattened[2], flattened[3] = nil, nil, nil
		flattened[4], flattened[5], flattened[6], flattened[7] = nil, nil, nil, nil
		flattened[8], flattened[9], flattened[10] = nil, nil, nil
		flattened[12], flattened[13] = nil, nil

		// Re-import the data square.
		eds, err := ImportExtendedDataSquare(flattened, codec, NewDefaultTree)
		if err != nil {
			t.Errorf("ImportExtendedDataSquare failed: %v", err)
		}

		err = eds.Repair(rowRoots, colRoots)
		if err != nil {
			t.Errorf("unexpected err while repairing data square: %v, codec: :%s", err, codec.Name())
		} else {
			assert.Equal(t, original.GetCell(0, 0), bytes.Repeat([]byte{1}, shareSize))
			assert.Equal(t, original.GetCell(0, 1), bytes.Repeat([]byte{2}, shareSize))
			assert.Equal(t, original.GetCell(1, 0), bytes.Repeat([]byte{3}, shareSize))
			assert.Equal(t, original.GetCell(1, 1), bytes.Repeat([]byte{4}, shareSize))
		}
	})

	// Verify that an EDS returns an error when there are too many erasures
	t.Run("Unrepairable", func(t *testing.T) {
		flattened := original.Flattened()
		flattened[0], flattened[2], flattened[3] = nil, nil, nil
		flattened[4], flattened[5], flattened[6], flattened[7] = nil, nil, nil, nil
		flattened[8], flattened[9], flattened[10] = nil, nil, nil
		flattened[12], flattened[13], flattened[14] = nil, nil, nil

		// Re-import the data square.
		eds, err := ImportExtendedDataSquare(flattened, codec, NewDefaultTree)
		if err != nil {
			t.Errorf("ImportExtendedDataSquare failed: %v", err)
		}

		err = eds.Repair(rowRoots, colRoots)
		if err != ErrUnrepairableDataSquare {
			t.Errorf("did not return an error on trying to repair an unrepairable square")
		}
	})
}
```

Reply:
gotcha, i thought there might be a use case of multiple test cases in the future. Will remove it